Quests/Kanto/CascadeBadgeQuest.lua

```diff
@@ -26,28 +30,39 @@ function CascadeBadgeQuest:new()
 end
 
 function CascadeBadgeQuest:isDoable()
-	if not hasItem("Cascade Badge") and self:hasMap()
-	then
+	if self:hasMap() then
```

Comment:
`hasItem("Cascade Badge")` is relevant there. The player will have to come back to this map later, the Cascade Badge is the only information that will allow us to select the right map.
The next Quest should start from the Cerulean Arena, even so this quest stop at Route 5.